### PR TITLE
Implement more methods for horse inventories

### PR DESCRIPTION
**Pull request by Lulu13022002**
This depends on my previous PR to avoid conflict.

Implement some missing methods (set/getContents and isEmpty) to take in account the armor slot. This fixes by extension other api using those methods such as addItem and more.

### CODE REVIEW

**Review by Machine-Maker — APPROVED** [review]
I originally modeled the changes off of CraftResultInventory which *doesn't* override these various methods. Perhaps it should be changed there as well, but that can be in a different PR.

File: patches/server/1031-fix-horse-inventories.patch [hunk 1]
```diff
@@ -74,6 +74,42 @@ index 4946da593713f4d11d88ac1bb68a089f2f6d5ae0..abef7f23361e6c5d18243dd18439ffd1
...
++        }
++
++        return items;
++    }
++
++    @Override
++    public void setContents(ItemStack[] items) {
++        com.google.common.base.Preconditions.checkArgument(items.length <= this.getSize(), "Invalid inventory size (%s); expected %s or less", items.length, this.getSize());
++
++        this.setSaddle(items[net.minecraft.world.entity.animal.horse.AbstractHorse.INV_SLOT_SADDLE]);
```
Comment (lynxplay) [inline patches/server/1031-fix-horse-inventories.patch]: These two setters seems wrong. The passed `items` array may be empty. Or only include a single item.
These two lines would IOB.

Comment (Lulu13022002) [reply patches/server/1031-fix-horse-inventories.patch]: yes i found a similar issue on the getter.

**Comment by Lulu13022002** [comment]
Yes you already have a pr for other inventories (https://github.com/PaperMC/Paper/pull/10721/files) however you only override the getter for some reason.

File: patches/server/1031-fix-horse-inventories.patch [hunk 2]
```diff
@@ -74,6 +74,42 @@ index 4946da593713f4d11d88ac1bb68a089f2f6d5ae0..abef7f23361e6c5d18243dd18439ffd1
...
++    }
++
++    @Override
++    public void setContents(ItemStack[] items) {
++        com.google.common.base.Preconditions.checkArgument(items.length <= this.getSize(), "Invalid inventory size (%s); expected %s or less", items.length, this.getSize());
++
++        ItemStack saddle = items.length > net.minecraft.world.entity.animal.horse.AbstractHorse.INV_SLOT_SADDLE ? items[net.minecraft.world.entity.animal.horse.AbstractHorse.INV_SLOT_SADDLE] : null;
++        ItemStack armor = items.length > net.minecraft.world.inventory.HorseInventoryMenu.SLOT_BODY_ARMOR ? items[net.minecraft.world.inventory.HorseInventoryMenu.SLOT_BODY_ARMOR] : null;
++
++        this.setSaddle(saddle);
```
Comment (lynxplay) [inline patches/server/1031-fix-horse-inventories.patch]: org.apache.commons.lang3.ArrayUtils#get would make this a lot nicer to look at.

**Pull request merged**